packages/connect-conformance/src/web/client.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ClientCompatResponse } from "../gen/connectrpc/conformance/v1/client_compat_pb.js";
+import { Buffer } from "node:buffer";
+import { remote } from "webdriverio";
+import type { RemoteOptions } from "webdriverio";
+import * as esbuild from "esbuild";
+import { parseArgs } from "node:util";
+import { readSizeDelimitedBuffers } from "../protocol.js";
+
+const { values: flags } = parseArgs({
+  args: process.argv.slice(2),
+  options: {
+    browser: { type: "string", default: "chrome" },
+  },
+});
+
+export async function run() {
+  let capabilities: RemoteOptions["capabilities"] = {
+    acceptInsecureCerts: true,
+  };
+  switch (flags.browser) {
+    case "chrome":
+    case undefined:
+      capabilities = {
+        ...capabilities,
+        browserName: "chrome",
+        "goog:chromeOptions": {
+          args: ["--disable-gpu", "--headless"],
+        },
+      };
+      break;
+    case "firefox":
+      capabilities = {
+        ...capabilities,
+        browserName: "firefox",
+        "moz:firefoxOptions": {
+          args: ["-headless"],
+        },
+      };
+      break;
+    case "safari":
+      capabilities = {
+        ...capabilities,
+        browserName: "safari",
+      };
+      break;
+    default:
+      throw new Error(`Unsupported browser: ${flags.browser}`);
+  }
+  const browser = await remote({
+    // webdriverio prints all the logs to stdout, this will interfere with the conformance test output
+    // so we set the log level to silent
+    logLevel: "silent",
```

Comment:
I'm surprised because that's exactly what @smaye81 and I tried, and it wasn't effective. 

Was there an issue in our I/O functions? Because I see that readSizeDelimitedBuffers has changed over [readReqBuffers](https://github.com/connectrpc/conformance/blob/926b5346de2357ebfe8dceb4ac46f128c8013e5d/testing/grpcwebclient/grpcwebclient.ts#L83C17-L83C31).

One downside of just silencing the log is that it can contain helpful information. If tests fail, it will be very time-consuming to get to the bottom of an issue if the crucial pice of information is in /dev/null. Can you take a look at redirecting the log to a git-ignored file? If there is no simple solution, let's at least add add a TODO.

Reply:
I think I know why it didn't work. There is a simpler package called `webdriver` which also accepts the same options but after reading the source code I realized it doesn't set the log values anywhere. I looked at the wrapper package `webdriverio` which does set the log level. 

I tried redirecting the logs using the `outputDir` option but that only redirects the driver logs and logs everything else to stdout.

Regarding debugging, if it is regarding setting up the browser we can just run the code directly by commenting this line and it'll pause for input right after the browser setup is done. If we remove the headless options, we can use the browser's console and inspect tab to debug.

I'll add a TODO here and also address the headless comment.